### PR TITLE
support to get registry config from cluster env

### PR DESCRIPTION
**Pull request by Stevent-fei**
### Describe what this PR does / why we need it
1. add custom domain
2. add custom registry username
3. add custom registry password

### CODE REVIEW

**Comment by kakaZhou719** [comment]
would you mind change your pr title to `support to get registry config from cluster env` ? ,  add your test clusterfile and show us the result.

File: pkg/cluster-runtime/installer.go [hunk 1]
```diff
@@ -70,14 +71,30 @@ func NewInstaller(infraDriver infradriver.InfraDriver, runtimeConfig RuntimeConf
...
 	installer.RegistryConfig.LocalRegistry = &registry.LocalRegistry{
 		DataDir:      filepath.Join(infraDriver.GetClusterRootfsPath(), "registry"),
 		InsecureMode: false,
 		Cert:         &registry.TLSCert{},
 		DeployHost:   infraDriver.GetHostIPListByRole(common.MASTER)[0],
 		Registry: registry.Registry{
-			Domain: registry.DefaultDomain,
+			Domain: domain,
 			Port:   registry.DefaultPort,
 			Auth:   &registry.Auth{},
```
Comment (kakaZhou719) [inline pkg/cluster-runtime/installer.go:99]: Auth:auth.

File: pkg/cluster-runtime/installer.go [hunk 2]
```diff
@@ -70,14 +71,30 @@ func NewInstaller(infraDriver infradriver.InfraDriver, runtimeConfig RuntimeConf
 		return nil, err
 	}
 
+	auth := registry.Auth{}
```
Comment (kakaZhou719) [inline pkg/cluster-runtime/installer.go:74]: you could new default registry config first, and then to do overwrite.

File: pkg/cluster-runtime/installer.go [hunk 3]
```diff
@@ -70,17 +70,31 @@ func NewInstaller(infraDriver infradriver.InfraDriver, runtimeConfig RuntimeConf
...
+	// todo maybe we can support custom registry config later
+
+	domainEnv := infraDriver.GetClusterEnv()
+	var RegistryConfig = registry.Registry{
+		Domain: registry.DefaultDomain,
+		Port:   registry.DefaultPort,
+		Auth:   &registry.Auth{},
+	}
+	if value := domainEnv["RegistryDomain"]; value != nil {
+		RegistryConfig.Domain = fmt.Sprint(value)
```
Comment (kakaZhou719) [inline pkg/cluster-runtime/installer.go:82]: rename `value` to `domain`

Comment (kakaZhou719) [inline pkg/cluster-runtime/installer.go:82]: just use string() to do it.

**Review by VinceCui — APPROVED** [review]
lgtm

**Pull request merged**